### DISCUSSION TITLE
How should librespot passwords be encoded?

### DISCUSSION THREAD

**Discussion by GeorgeHahn** [snapcast/snapcast#1245] [Q&A] [answered]
Hi,

I'm having some trouble setting up a librespot stream. My password contains some special characters and authentication is failing. From reading the source, I see that they are at least URL encoded. That said, authentication fails with my URL-encoded password.

Can you share details on how passwords need to be encoded?

Thank you!

**Comment by chiefy** [comment]
@GeorgeHahn I had this very issue this weekend when I was trying to setup librespot. I could not get it to work. My suggestion would be to use a password generator where you can specify not to use special characters.

**Reply by GeorgeHahn** [reply]
Agreed, that seems to be the best workaround. I manually launched librespot in pipe mode for a while and ended up switching to a completely different setup.

**Reply by chiefy** [reply, marked as answer]
I actually ended up figuring it out - you can no longer use password auth in librespot. I ended up using this tool and passing the creds to librespot that way, works now. https://github.com/dspearson/librespot-auth
